lib/uplink/ext/main.go

```diff
@@ -0,0 +1,8 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+//go:generate go build -o uplink-cgo.so -buildmode=c-shared .
```

Comment:
Creating `.so` with go generate feels very wrong to me. It probably should be part of a makefile or some other approach. I think this is the reason why it was also globbering the source directory.

Reply:
I don't have a preference on how it gets built. Currently a make target (`libuplink`) is calling `go generate` on this file.

Reply:
What are you talking about with the source directory?